Sources/TSCBasic/PathShims.swift

```diff
@@ -24,20 +24,21 @@ import Foundation
 public func resolveSymlinks(_ path: AbsolutePath) throws -> AbsolutePath {
 #if os(Windows)
     let handle: HANDLE = path.pathString.withCString(encodedAs: UTF16.self) {
-        CreateFileW($0, GENERIC_READ, DWORD(FILE_SHARE_READ), nil,
+        CreateFileW($0, GENERIC_READ, DWORD(FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE), nil,
                     DWORD(OPEN_EXISTING), DWORD(FILE_FLAG_BACKUP_SEMANTICS), nil)
     }
     if handle == INVALID_HANDLE_VALUE { return path }
     defer { CloseHandle(handle) }
 
     let dwLength: DWORD =
-        GetFinalPathNameByHandleW(handle, nil, 0, DWORD(FILE_NAME_NORMALIZED))
+        GetFinalPathNameByHandleW(handle, nil, 0, DWORD(VOLUME_NAME_DOS))
     return try withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(dwLength)) {
         guard GetFinalPathNameByHandleW(handle, $0.baseAddress!, DWORD($0.count),
-                                        DWORD(FILE_NAME_NORMALIZED)) == dwLength - 1 else {
+                                        DWORD(VOLUME_NAME_DOS)) == dwLength - 1 else {
             throw FileSystemError(.unknownOSError, path)
         }
-        let path = String(decodingCString: $0.baseAddress!, as: UTF16.self)
+        // When using `VOLUME_NAME_DOS`, the returned path uses `\\?\`.
+        let path = String(decodingCString: $0.baseAddress!.advanced(by: 4), as: UTF16.self)
```

Comment:
Do we want an assertion that the prefix is present?

Reply:
I was following what you did in https://github.com/swiftlang/swift-foundation/pull/639. I guess we should also add the check there.

Reply:
Making the corresponding change to swift-foundation here: https://github.com/swiftlang/swift-foundation/pull/974